.github/workflows/scanner-offline-bundle-update.yaml

```diff
@@ -1,37 +1,48 @@
 name: Scanner release offline vulnerability bundle update
+
 on:
   schedule:
     - cron: '0 */3 * * *'
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
   workflow_dispatch:
 
 jobs:
-  read-release-versions:
+  generate-version-matrix:
     runs-on: ubuntu-latest
+    if: >
+      github.event_name == 'schedule' ||
+      (github.event_name == 'pull_request' &&
+       contains(github.event.pull_request.labels.*.name, 'pr-update-scanner-vulns'))
     outputs:
       matrix: ${{ steps.output-matrix.outputs.matrix }}
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
 
-    - name: Generate matrix JSON
+    - name:
       id: output-matrix
       run: |
         EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
         echo "matrix<<$EOF" >> "$GITHUB_OUTPUT"
-        # Extract X.Y versions, ensuring uniqueness
-        grep -oE '^[0-9]+\.[0-9]+' scanner/updater/version/RELEASE_VERSION | sort -u | jq -Rs '{ versions: ( sub("\n$"; "")|split("\n") ) }' | tee -a "$GITHUB_OUTPUT"
+        .github/workflows/scripts/scanner-offline-bundle-version-matrix.sh | tee -a "$GITHUB_OUTPUT"
         echo "$EOF" >> "$GITHUB_OUTPUT"
```

Comment:
this generates `4.4.0`, `4.5.0`, and `v1` with all the associated versions. `v1` encompasses all of those versions. Are we effectively creating/uploading everything twice?

Reply:
Here's the output as of today:

```json
.github/workflows/scripts/scanner-offline-bundle-version-matrix.sh
{"versions": [
  {
    "vulnerability_version": "v1",
    "supported_releases": "4.4.0 4.4.1 4.4.2 4.4.3 4.4.4 4.4.5 4.5.0 4.5.1 4.5.2"
  }
,
  {
    "vulnerability_version": "4.4.0",
    "supported_releases": "4.4.0 4.4.1 4.4.2 4.4.3 4.4.4 4.4.5 "
  }
,
  {
    "vulnerability_version": "4.5.0",
    "supported_releases": "4.5.0 4.5.1 4.5.2 "
  }
]}
```

Here's the explanation why (from the Description):

> For each execution in the matrix, create a corresponding scanner-v4-defs-<schema version>.zip bundle; 4.[45].x releases will be created as well, to ensure backward compatibility.

That means:

- We build one offline definition for v1, that will be used for all versions 4.6 onward.
- We build 4.50 and 4.4.0 offline definitions, that will be used on 4.5.* and 4.4.* releases, respectively.

See https://github.com/stackrox/scanner/pull/1643 on how these definition bundles will be used to create the offline bundles for each release. That might also clarify the reasoning behind this change, if the description and this comment is not sufficient for you.

Reply:
`supported_releases` will allow Central to validate if this V4 offline definition supports its version tag, when uploading the offline bundle. It means if the user uploads a bundle that does not support the Central release, then the upload will fail. This ensures only supported bundles are available for download from Scanner.